packages/paypal-express-integration/src/paypal-express-button-strategy.ts

```diff
@@ -0,0 +1,181 @@
+    ): Promise<void> {
+        const state = this.paymentIntegrationService.getState();
+        const paymentMethod = state.getPaymentMethodOrThrow(options.methodId);
+
+        if (!paymentMethod.config.merchantId) {
+            throw new MissingDataError(MissingDataErrorType.MissingPaymentMethod);
+        }
+
+        const paypalSDK = await this.paypalExpressScriptLoader.loadPaypalSDK(
+            paymentMethod.config.merchantId,
+        );
+
+        this.renderButton(options, paypalSDK);
+    }
+
+    private renderButton(
+        options: CheckoutButtonInitializeOptions & WithPaypalExpressButtonInitializeOptions,
+        paypalSDK: PaypalSDK,
+    ) {
+        const state = this.paymentIntegrationService.getState();
+        const paymentMethod = state.getPaymentMethodOrThrow(options.methodId);
+
+        if (!paymentMethod.config.merchantId) {
+            throw new MissingDataError(MissingDataErrorType.MissingPaymentMethod);
+        }
+
+        const paypalOptions = options.paypal;
+        const merchantId = paymentMethod.config.merchantId;
+        const env = paymentMethod.config.testMode ? 'sandbox' : 'production';
+        const clientToken: PaypalClientToken = { [env]: paypalOptions.clientId };
+
+        const fundingCreditOption = paypalSDK.FUNDING.CREDIT || 'credit';
+        const allowedSources = paypalOptions.allowCredit ? [fundingCreditOption] : [];
+        const disallowedSources = !paypalOptions.allowCredit ? [fundingCreditOption] : [];
+
+        return paypalSDK.Button.render(
+            {
+                env,
+                client: clientToken,
+                commit: paypalOptions.shouldProcessPayment,
+                funding: {
+                    allowed: allowedSources,
+                    disallowed: disallowedSources,
+                },
+                style: this.getStyle(paypalOptions, paymentMethod.id),
+                payment: (_, actions) =>
+                    this.setupPayment(merchantId, actions, paypalOptions.onPaymentError),
+                onAuthorize: (data, actions) =>
+                    this.tokenizePayment(data, options.methodId, actions, paypalOptions.shouldProcessPayment),
+            },
+            options.containerId,
+        );
+    }
+
+    deinitialize(): Promise<void> {
+
+        return Promise.resolve();
+    }
+
+    private getStyle(paypalOptions: PaypalExpressButtonInitializeOptions, paymentMethodId: string) {
+       return  {
+            shape: PaypalButtonStyleShapeOption.RECT,
+        ...pick(
+            paypalOptions.style,
+            'layout',
+            'color',
+            'label',
+            'shape',
+            'tagline',
+            'fundingicons',
+        ),
+            size:
+            paymentMethodId === 'paypalexpress' &&
+            paypalOptions.style?.size === 'small'
+                ? PaypalButtonStyleSizeOption.RESPONSIVE
+                : paypalOptions.style?.size,
+        }
+    }
```

Comment:
I think `paymentMethodId` will always be equal to `'paypalexpress'`. This method can be simplified:

```suggestion
    private getStyle(style?: Omit<PaypalStyleOptions, 'height'>): PaypalStyleOptions {
        const {
            color,
            fundingicons,
            label,
            layout,
            shape,
            size,
            tagline,
        } = style || {};
    
       return  {
            color,
            fundingicons,
            label,
            layout,
            shape: shape || PaypalButtonStyleShapeOption.RECT,
            size: size === 'small'
                ? PaypalButtonStyleSizeOption.RESPONSIVE
                : size,
            tagline,
        }
    }
```

Reply:
fixed